public/app/features/dashboard-scene/serialization/transformSaveModelSchemaV2ToScene.ts

```diff
@@ -0,0 +1,610 @@
+  defaultQueryVariableKind,
+  defaultTextVariableKind,
+  GroupByVariableKind,
+  IntervalVariableKind,
+  PanelKind,
+  QueryVariableKind,
+  TextVariableKind,
+} from '@grafana/schema/src/schema/dashboard/v2alpha0/dashboard.gen';
+
+import { addPanelsOnLoadBehavior } from '../addToDashboard/addPanelsOnLoadBehavior';
+import { DashboardAnnotationsDataLayer } from '../scene/DashboardAnnotationsDataLayer';
+import { DashboardControls } from '../scene/DashboardControls';
+import { DashboardDataLayerSet } from '../scene/DashboardDataLayerSet';
+import { DashboardDatasourceBehaviour } from '../scene/DashboardDatasourceBehaviour';
+import { registerDashboardMacro } from '../scene/DashboardMacro';
+import { DashboardReloadBehavior } from '../scene/DashboardReloadBehavior';
+import { DashboardScene } from '../scene/DashboardScene';
+import { DashboardScopesFacade } from '../scene/DashboardScopesFacade';
+import { panelMenuBehavior } from '../scene/PanelMenuBehavior';
+import { PanelNotices } from '../scene/PanelNotices';
+import { PanelTimeRange } from '../scene/PanelTimeRange';
+import { AngularDeprecation } from '../scene/angular/AngularDeprecation';
+import { DashboardGridItem } from '../scene/layout-default/DashboardGridItem';
+import { DefaultGridLayoutManager } from '../scene/layout-default/DefaultGridLayoutManager';
+import { setDashboardPanelContext } from '../scene/setDashboardPanelContext';
+import { preserveDashboardSceneStateInLocalStorage } from '../utils/dashboardSessionState';
+import { DashboardInteractions } from '../utils/interactions';
+import { getDashboardSceneFor, getIntervalsFromQueryString } from '../utils/utils';
+
+import { SnapshotVariable } from './custom-variables/SnapshotVariable';
+import {
+  transformCursorSyncV2ToV1,
+  transformSortVariableToEnumV1,
+  transformValueMappingsToV1,
+  transformVariableHideToEnumV1,
+  transformVariableRefreshToEnumV1,
+} from './transformToV1TypesUtils';
+
+const DEFAULT_DATASOURCE = 'default';
+
+type TypedVariableModelv2 =
+  | QueryVariableKind
+  | TextVariableKind
+  | ConstantVariableKind
+  | DatasourceVariableKind
+  | IntervalVariableKind
+  | CustomVariableKind
+  | GroupByVariableKind
+  | AdhocVariableKind;
+
+export function transformSaveModelSchemaV2ToScene(dashboard: DashboardV2Spec): DashboardScene {
+  const annotationLayers = dashboard.annotations.map((annotation) => {
+    return new DashboardAnnotationsDataLayer({
+      key: uniqueId('annotations-'),
+      query: annotation.spec,
+      name: annotation.spec.name,
+      isEnabled: Boolean(annotation.spec.enable),
+      isHidden: Boolean(annotation.spec.hide),
+    });
+  });
+
+  const dashboardScene = new DashboardScene({
+    description: dashboard.description,
+    editable: dashboard.editable,
+    preload: dashboard.preload,
+    id: dashboard.id,
+    isDirty: false,
+    links: dashboard.links,
+    meta: {},
+    tags: dashboard.tags,
+    title: dashboard.title,
+    uid: dashboard.id?.toString(),
```

Comment:
The uid will come from k8s metadata, that will be resource's `metadata.name`. It means, that the entire resource needs to be passed as the argument for transformation (`DashboardWithAccessInfo` type).